README.md

```diff
@@ -1 +1,429 @@
+  local data = {5, { }}
+  ```
+- Spaces after commas
+  ```lua
+  -- Good
+  myFunc( 10, { 3, 5 } )
+  
+  -- Bad
+  myFunc( 10,{ 3,5 } )
+  ```
+- Indentation should be done with 4 spaces
+  ```lua
+  -- Good
+  if cond then
+    myFunc()
+  end
+
+  -- Bad
+  if cond then
+  	myFunc()
+  end
+  ```
+- No spaces inside square brackets
+   ```lua
+   -- Good
+  local val = tab[5] + tab[3]
+  local val2 = tab[5 * 3]
+
+  -- Bad
+  local val = tab[ 5 ] + tab[ 3 ]
+  local val2 = tab[ 5 * 3 ]
+  ```
+- Single space after comment operators
+  ```lua
+  -- Good
+  -- This is a good comment
+  
+  -- Bad
+  --This comment is too close to the operator
+  ```
+
+## Newlines
+- Never have more than 2 newlines
+  ```lua
+  -- Good
+  local config = GM.Config
+
+
+  function GM:Think()
+    -- do thing
+  end
+
+  -- Bad
+  local config = GM.Config
+
+
+
+  function GM:Think()
+    -- do thing
+  end
+  ```
+- Top level blocks should have either 1 or 2 newlines between them
+  ```lua
+  -- Good
+  local config = GM.Config
+
+
+  function GM:Think()
+    -- do thing
+  end
+
+  -- Bad
+  local config = GM.Config
+  function GM:Think()
+    -- do thing
+  end
+  ```
+- Non top level blocks/lines should never have more than 1 newline between them
+  ```lua
+  -- Good
+  function test()
+    local a = 3
+
+	print( a )
+  end
+
+  -- Bad
+  function test()
+    local a = 3
+
+
+	print( a )
+  end
+  ```
+- Returns should have 1 newline before them unless the codeblock is only 1 line
+  ```lua
+  -- Good
+  function test()
+    local a = 3
+
+	return a
+  end
+
+  function test2()
+    return 3
+  end
+
+  -- Bad
+  function test()
+    local a = 3
+	return a
+  end
+  ```
+- Code should be split into managable chunks using a single new line
+  ```lua
+  -- Good
+  function CFCNotifications.resolveFilter( filter )
+    if type( filter ) == "Player" then
+      filter = { filter }
+    end
+  
+    if type( filter ) == "table" then
+      filter = fWrap( filter )
+    end
+  
+    filter = filter or player.GetAll
+    local players = filter()
+  
+    if type( players ) == "Player" then
+      players = { players }
+    end
+  
+    if not players then players = player.GetAll() end
+  
+    return players
+  end
+
+  -- Bad
+  function CFCNotifications.resolveFilter( filter )
+    if type( filter ) == "Player" then
+      filter = { filter }
+    end
+    if type( filter ) == "table" then
+      filter = fWrap( filter )
+    end
+    filter = filter or player.GetAll
+    local players = filter()
+    if type( players ) == "Player" then
+      players = { players }
+    end
+    if not players then players = player.GetAll() end
+    return players
+  end
+  ```
+
+## Gmod lua additions
+- Operators added by gmod such as `&&` `||` `!` `!=` should never be used
+  ```lua
+  -- Good
+  if a ~= b and not b then end
+  
+  -- Bad
+  if a != b && !b then end
+  ```
+- GMod style comments should never be used `/* */` and `//`
+  ```lua
+  -- Good
+  /*
```

Comment:
this should be in the bad part, not good

Reply:
who are you to tell me what to do